pandas/core/arrays/datetimes.py

```diff
@@ -309,6 +309,8 @@ def __init__(self, values, dtype=_NS_DTYPE, freq=None, copy=False):
                 "ndarray, or Series or Index containing one of those."
             )
             raise ValueError(msg.format(type(values).__name__))
+        if values.ndim != 1:
```

Comment:
is this actually hit in the test suite?

Reply:
Yes, plus a dedicated test is added by this PR.